src/hevm/src/EVM/Concrete.hs

```diff
@@ -98,21 +101,19 @@ sliceMemory o s m =
 
 writeMemory :: ByteString -> Word -> Word -> Word -> ByteString -> ByteString
 writeMemory bs1 (C _ n) (C _ src) (C _ dst) bs0 =
-  if src > num (BS.length bs1)
-  then
-    let
-      (a, b) = BS.splitAt (num dst) bs0
-      c      = BS.replicate (num n) 0
-      b'     = BS.drop (num n) b
-    in
-      a <> c <> b'
-  else
-    let
-      (a, b) = BS.splitAt (num dst) bs0
-      c      = BS.take (num n) (BS.drop (num src) bs1)
-      b'     = BS.drop (num n) b
-    in
-      a <> BS.replicate (num dst - BS.length a) 0 <> c <> b'
+  let
+    (a, b) = BS.splitAt (num dst) bs0
+    a'     = BS.replicate (num dst - BS.length a) 0
+    -- sliceMemory should work for both cases, but we are using 256 bit
+    -- words, whereas ByteString is only defined up to 64 bit. For large n,
+    -- src, dst this will cause problems (often in GeneralStateTests).
+    -- Later we could reimplement ByteString for 256 bit arguments.
```

Comment:
@livnev ^ n.b. Data.ByteString limitations with large numbers. In practice the size limit was somewhat higher than int64, but well below 256.

This gives an interesting "feature" of Data.ByteString is that `bs == BS.drop n bs` for sufficiently large `n`.